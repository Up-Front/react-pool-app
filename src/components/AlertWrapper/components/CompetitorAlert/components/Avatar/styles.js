import styled from 'styled-components';

export const AvatarImage = styled.img`
  position: relative;
  top: 0;
  width: 30vw;
  height: 30vw;
  border-radius: 50%;
  border: 1.3vw solid grey;
`;
