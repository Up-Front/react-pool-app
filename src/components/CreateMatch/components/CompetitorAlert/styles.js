import styled, { injectGlobal } from 'styled-components';

injectGlobal`
  @font-face {
    font-family: 'Permanent Marker';
    font-style: normal;
    font-weight: 400;
    src: local('Permanent Marker Regular'), local('PermanentMarker-Regular'), url(https://fonts.gstatic.com/s/permanentmarker/v7/Fh4uPib9Iyv2ucM6pGQMWimMp004La2Cf5b6jlg.woff2) format('woff2');
    unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }
`;

export const AlertWrapper = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1000;

  & > div {
    display: flex;
    align-items: center;
    justify-content: center;

    & > div {
      flex: 1;
      position: absolute;
    }

    & > div:last-child {
      width: 100vw;
    }
  }
`;

export const WinnerWarningCenter = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 30vw;
`;

export const AvatarImage = styled.img`
  position: absolute;
  top: 0;
  width: 30vw;
  height: 30vw;
  border-radius: 50%;
`;

export const LeaderIcon = styled.div`
  position: relative;
  width: 15vw;
  height: 15vw;
  transform: rotate(-30deg);

  &:after {
    content: 'ðŸ‘‘';
    position: relative;
    top: -5vw;
    left: 5vw;
    font-size: 8vw;
  }
`;

export const BoomTextBase = styled.h2`
  position: absolute;
  top: -7vw;
  transform: rotate(-5deg);
  width: 100vw;
  text-align: center;
  font-family: 'Permanent Marker';
  font-size: 10vw;
`;

export const BoomText = styled(BoomTextBase)`
  background: linear-gradient(
    to bottom,
    #e8bf2e 0%,
    #e58500 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  z-index: 1;
`;

export const BoomTextShadow = styled(BoomTextBase)`
  opacity: 0.4;
  text-shadow: 4px 4px 2px rgba(0, 0, 0, 1);
`;
