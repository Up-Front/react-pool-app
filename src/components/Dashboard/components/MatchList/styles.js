import styled, { injectGlobal } from 'styled-components';

injectGlobal`
  .remove-enter {
    max-height: 0;
  }
  .remove-enter-active {
    max-height: 100px;
    transition: all 500ms ease-in;
  }
  .remove-exit {
    max-height: 100px;
  }
  .remove-exit-active {
    max-height: 0;
    transition: all 500ms ease-in;
  }
`;

export const MatchListWrapper = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  overflow: hidden;
`;
