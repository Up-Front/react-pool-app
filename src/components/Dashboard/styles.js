import styled from 'styled-components';

export const FloatButton = styled.button`
  position: absolute;
  bottom: 20px;
  right: 20px;

  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 0;

  font-size: 24px;
  font-weight: bold;
  line-height: 24px;
`;
