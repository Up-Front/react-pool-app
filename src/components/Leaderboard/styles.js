import styled from 'styled-components';

export const ListUser = styled.div`
  transform: scale(1);
  transition: all 500ms ease-in;
  ${props =>
    props.changedRank
      ? `
      transform: scale(0); 
  background-color: green;
  `
      : ``};

  ${props =>
    props.showChangedRank
      ? `
 
  transform: scale(0);
  background-color: yellow;
  
  `
      : ``};
`;
