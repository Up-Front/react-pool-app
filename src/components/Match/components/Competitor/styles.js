import styled from 'styled-components';
import constants from './../../../shared/constants';

export const CompetitorWrapper = styled.div`
  flex: 1;
`;

export const CompetitorLine = styled.div`
  display: flex;
  flex-direction: ${props =>
    props.align === constants.ALIGNRIGHT ? 'row-reverse' : 'row'};
  align-items: center;
  width: 100%;
`;

export const CompetitorAvatar = styled.img`
  margin: 5px;
  width: 40px;
  height: 40px;
  border-radius: 50%;
`;

export const CompetitorName = styled.div`
  flex: 1;
  color: ${props => (props.hasVote ? 'green' : '')};
  text-align: ${props =>
    props.align === constants.ALIGNRIGHT ? 'right' : 'left'};
`;
