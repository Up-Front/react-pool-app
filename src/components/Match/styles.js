import styled from 'styled-components'

export const MatchWrapper = styled.li`
    margin: 2px 0;
    padding: 10px;
    width: 100%;
    border: 1px solid white;

    background-color: ${props => props.contested ? 'red' : ''};
`;