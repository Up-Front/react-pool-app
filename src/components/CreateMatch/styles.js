import styled from 'styled-components';
import { iconStyle } from './../shared/styles';

export const SelectOpponent = styled.section`
  position: absolute;
  z-index: 100;
  width: 100%;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.7);
  display: ${props => (props.show ? 'block' : 'none')};
`;

export const FloatButton = styled.button`
  position: absolute;
  bottom: 20px;
  right: 20px;

  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 0;

  font-size: 24px;
  font-weight: bold;
  line-height: 24px;

  z-index: 998;
`;

export const SearchField = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  max-height: 60px;
  overflow: hidden;
`;

export const SearchResult = styled.div`
  position: relative;
  width: 100%;
`;

export const SearchClear = styled.button`
  position: absolute;
  top: 16px;
  right: 10px;
  border: 0;
  color: white;
  cursor: pointer;
  background-color: transparent;
  &:before {
    ${iconStyle('times')};
  }
`;
