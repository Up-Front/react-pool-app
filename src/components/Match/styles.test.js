import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import { MatchWrapper } from './styles';

describe('MatchWrapper component', () => {
    test('renders normally', () => {
        const wrapper = shallow(<MatchWrapper />);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders contested', () => {
        const wrapper = shallow(<MatchWrapper contested />);
        expect(wrapper).toHaveStyleRule('background-color', 'red');
    });
});

