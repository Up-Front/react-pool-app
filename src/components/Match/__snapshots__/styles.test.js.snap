// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchWrapper component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.li />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa gooclE",
    },
    "ref": null,
    "rendered": null,
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa gooclE",
      },
      "ref": null,
      "rendered": null,
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
